ticketauth/src/main/kotlin/dk/ufst/ticketauth/AuthResult.kt

```diff
@@ -1,7 +1,21 @@
 package dk.ufst.ticketauth
 
-enum class AuthResult {
-    SUCCESS,
-    CANCELLED_FLOW,
-    ERROR
+import androidx.activity.result.ActivityResult
+import dk.ufst.ticketauth.authcode.AuthorizationError
+import dk.ufst.ticketauth.authcode.RedirectUriParser
+
+sealed interface AuthResult {
+    object SUCCESS : AuthResult
+    object CANCELLED_FLOW : AuthResult
+    data class ERROR(val reason: ErrorCause) : AuthResult
```

Comment:
I have kept the uppercase underscore-separated names style, but we can change it to upper camel case.